client/components/language-picker/modal.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * External dependencies
+ */
+import React, { ReactNode, useState } from 'react';
+import { Dialog } from '@automattic/components';
+import { useI18n } from '@automattic/react-i18n';
+import LanguagePicker, { Language, createLanguageGroups } from '@automattic/language-picker';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Style dependencies
+ */
+import './modal.scss';
+
+type Props = {
+	languages: Language[];
+	/**
+	 * Fires when the modal is dismissed.
+	 */
+	onClose: () => void;
+	/**
+	 * Fires when the modal is dismissed with a language selection.
+	 */
+	onSelectLanguage: ( l: Language ) => void;
+	selectedLanguageSlug?: string;
+};
+
+const LanguagePickerModal = ( {
+	languages,
```

Comment:
Hi! 

Was there any reason not to fetch the localized languages from the `/i18n/language-names` endpoint (to enable localized searching)? 

Granted, there is a bit of [data mashing](https://github.com/Automattic/wp-calypso/blob/58b9a18753a486af0e13192e58d740dae1922c51/client/components/language-picker/modal.jsx#L153) to get the filtered results, but we're removing a feature so I just wanted to make sure it's intentional.

We're using a query component for that at the moment `<QueryLanguageNames />` in the [deleted modal picker file](https://github.com/Automattic/wp-calypso/pull/46328/files#diff-85febdb2d08b1d64ca197b57b3012332ac71b2a8d72d91f43be2257e136f7479L593).

Also asked here p9oQ9f-kT-p2#comment-548

Thanks!

cc @Automattic/i18n

Reply:
@ramonjd Do you have any pointers on how to do this from within Gutenboarding? We just need to get the response and the LanguagePicker component can already handle the result. I can see how to retrieve it in Calypso just using the existing query component, but that doesn't seem like it will work in Gutenboarding. I'm not at all familiar with how the data stores and all that stuff works.

Reply:
> they don't come back for the correct locale 

The endpoint doesn't check the user's default locale (assuming a user is logged in), but relies on the `_locale` query parameter.

https://public-api.wordpress.com/wpcom/v2/i18n/language-names?_locale=es

Otherwise the "localized" names will be in English.

So, I guess we'd have to pass the current locale to the selector (and indirectly to the [resolver](https://github.com/Automattic/wp-calypso/pull/46328/files#diff-f562d06e677604cc31cb94c3e9ca15a0e427645394d5bf86be5e1442e1245068R9)).

Normally, the locale will be stored in the URL fragment, e.g., `/new/es`, or `/new/language-modal/de`

I see that [getLocale](https://github.com/Automattic/wp-calypso/blob/5278b2e11d97c9937158c4890692c0b2d7258de3/client/landing/gutenboarding/components/locale-context/index.tsx#L163) exists to check the URL for the existence of a valid locale.

You can test with the Language picker enabled by adding the following flag: `/new/?flags=gutenboarding/language-picker`

Having said all this, one of our tasks for[ i18n Gutenboarding](https://github.com/Automattic/wp-calypso/issues/37665) was to enable the language picker, so if you get the data-store/i18n selector/resolver/apiFetch accepting a locale then we could look at passing the right one from Gutenboarding later if you run out of time.

> adding a new i18n store to data-stores

👍 That's what I was going to suggest - nice work. It's where most of the state-related stuff for Gutenboarding lives. I'll still ping @razvanpapadopol and @yansern, who know the Gutenboarding architecture back to front to make sure. :)

Reply:
The `_locale` query parameter should be added automatically to every REST request, based on what's currently in Redux `state.ui.language`. There is a [dedicated `wpcom` interceptor handler](https://github.com/Automattic/wp-calypso/blob/master/client/lib/wp/localization/index.js) for that. For the older WP.COM endpoints (as opposed to the newer WP REST ones), the parameter name is `locale`, without the `_`.

Gutenberg's `apiFetch` code has a [similar handler](https://github.com/WordPress/gutenberg/blob/master/packages/api-fetch/src/middlewares/user-locale.js) that, however, always adds `_locale=user`, telling the server to use the logged-in user's locale rather than default `en`. In logged-out mode, however, this does nothing.

 The Core REST code seems to not support anything other than `user`: https://github.com/WordPress/WordPress/blob/master/wp-includes/l10n.php#L143-L145 But an endpoint like `i18n/language-names` can check the `_locale` param on its own.

> Normally, the locale will be stored in the URL fragment, e.g., `/new/es`, or `/new/language-modal/de`

In most cases, the locale is determined from the current logged in user. But the locale from URL should make its way to the Redux and `i18n-calypso`, too. The Signup and Login handlers that use these locale suffixes should have code for that.

Hopefully the above will be helpful when debugging the `_locale` param. Calypso code should always add it, and Gutenberg code will probably need to add it explicitly.